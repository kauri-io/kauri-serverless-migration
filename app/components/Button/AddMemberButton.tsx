import * as React from "react";
import styled from "../../lib/styled-components";

interface IButtonStyleProps {
  bg: string;
  disabled: boolean;
  bgHover: string;
}

const AddMemberButton = styled<IButtonStyleProps, "img">("img")`
  height: 30px;
  width: 30px;
  cursor: pointer;
`;

interface IProps {
  handleClick?: () => void;
  onClick?: () => void;
  disabled?: boolean;
  bg?: string;
  bgHover?: string;
  fontWeight?: number;
  fontSize?: number;
}

const AddMemberButtonComponent: React.FunctionComponent<IProps> = ({
  bg = "primary",
  bgHover = "primaryDark",
  handleClick,
  onClick,
  disabled,
}) => (
  <AddMemberButton
    src="/static/images/icons/add-member-icon.png"
    disabled={Boolean(disabled)}
    onClick={onClick || handleClick}
    bg={bg}
    bgHover={bgHover}
  />
);

export default AddMemberButtonComponent;
