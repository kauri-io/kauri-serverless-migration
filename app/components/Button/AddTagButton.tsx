import * as React from "react";
import styled, { css } from "../../lib/styled-components";
import {
  space,
  fontSize as fontSizeSS,
  fontWeight as fontWeightSS,
} from "styled-system";

export const BaseButtonCss = css`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 26px;
  width: 80px;
  border-radius: 13px;
  cursor: pointer;
  opacity: ${({ disabled }) => (disabled ? "0.3" : "1")};
  background: transparent;
  border: 1px solid ${props => props.theme.colors[props.bg]};
  text-transform: uppercase;
  > svg,
  img {
    height: 16px;
    ${space};
  }
  ${fontWeightSS};
  ${fontSizeSS};
  color: ${props => props.theme.colors[props.color]};
  :hover {
    cursor: ${({ disabled }) => (disabled ? "not-allowed" : "pointer")};
  }
`;
interface IButtonProps {
  bg: string;
  fontWeight: number;
  fontSize: number;
}
const AddTagButton = styled<IButtonProps, "button">("button")`
  ${BaseButtonCss};
  :hover {
    border: 2px solid ${props => props.theme.colors[props.bg]};
  }
`;

interface IProps {
  handleClick?: () => void;
  onClick?: () => void;
  disabled?: boolean;
  bg: string;
  fontWeight?: number;
  fontSize?: number;
  space?: number;
  color?: string;
  text?: string;
  children?: Element;
}

export default ({
  bg = "primary",
  fontWeight = 700,
  fontSize = 0,
  color = "textPrimary",
  text = "Add Tag",
  handleClick,
  children,
  disabled,
}: IProps) => (
  <AddTagButton
    disabled={disabled}
    onClick={handleClick}
    bg={bg}
    color={color}
    fontSize={fontSize}
    fontWeight={fontWeight}
  >
    <img src="https://png.icons8.com/ios-glyphs/50/000000/plus-math.png" />
    {text || children}
  </AddTagButton>
);
