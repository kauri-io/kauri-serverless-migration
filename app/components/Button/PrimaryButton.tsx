import * as React from "react";
import styled, { css } from "styled-components";
import {
  space as spaceSS,
  fontSize as fontSizeSS,
  fontWeight as fontWeightSS,
  background as backgroundSS,
  color as colorSS,
  FontWeightProps,
  FontSizeProps,
  BackgroundProps,
  ColorProps,
  SpaceProps,
} from "styled-system";

export const BaseButtonCss = css`
  display: flex;
  align-items: center;
  justify-content: center;
  min-width: 136px;
  max-width: 150px;
  ${props =>
    typeof props.width !== "undefined" &&
    `width: ${props.width}; max-width: unset;`};
  height: 40px;
  border-radius: 4px;
  border: none;
  cursor: pointer;
  > svg,
  img {
    height: 16px;
    ${spaceSS};
  }
  text-transform: uppercase;
  opacity: ${({ disabled }) => (disabled ? "0.3" : "1")};
  ${fontWeightSS};
  ${fontSizeSS};
  ${backgroundSS};
  ${colorSS};
  :hover {
    cursor: ${({ disabled }) => (disabled ? "not-allowed" : "pointer")};
  }
`;

interface IPrimaryButtonSSProps {
  bgHover: string | undefined;
  width: string | undefined;
}

const PrimaryButton = styled<
  IPrimaryButtonSSProps &
    FontWeightProps &
    FontSizeProps &
    BackgroundProps &
    ColorProps &
    SpaceProps,
  "button"
>("button")`
  ${BaseButtonCss};
  :hover {
    background-color: ${props =>
      props.bgHover
        ? props.bgHover
        : props.theme && props.theme.colors && props.theme.colors.primaryDark};
  }
`;

interface IProps {
  icon?: React.ReactElement<any>;
  handleClick?: () => void;
  onClick?:void | (() => void);
  disabled?: boolean;
  type?: "button" | "submit" | "reset";
  bg?: string;
  bgHover?: string;
  fontWeight?: number;
  fontSize?: number;
  space?: number;
  color?: string;
  text?: string;
  width?: string;
  className?: string;
}

const PrimaryButtonComponent: React.SFC<IProps> = ({
  bg = "primary",
  bgHover,
  fontWeight = 700,
  fontSize = 0,
  space = 2,
  color = "white",
  type = "submit",
  onClick,
  handleClick,
  text = "",
  children,
  icon,
  disabled,
  width,
  className,
}) => (
  <PrimaryButton
    type={type}
    disabled={disabled}
    mr={space}
    onClick={onClick || handleClick}
    bg={bg}
    bgHover={bgHover}
    color={color}
    fontSize={fontSize}
    fontWeight={fontWeight}
    width={width}
    className={className}
  >
    {icon}
    {text || children}
  </PrimaryButton>
);

export default PrimaryButtonComponent;
