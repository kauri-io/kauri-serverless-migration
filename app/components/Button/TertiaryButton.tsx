import * as React from "react";
import styled from "../../lib/styled-components";
import {
  fontSize as fontSizeSS,
  fontWeight as fontWeightSS,
  color as colorSS,
  space as spaceSS,
  SpaceProps,
  FontSizeProps,
  FontWeightProps,
} from "styled-system";

const TertiaryButton = styled<
  { disabled: boolean } & SpaceProps & FontSizeProps & FontWeightProps,
  "button"
>("button")`
  display: flex;
  align-items: center;
  border: none;
  background-color: transparent;
  cursor: ${props => (props.disabled ? "not-allowed" : "pointer")};
  text-transform: uppercase;
  color: #fff;
  opacity: ${props => (props.disabled ? "0.5" : "1")};
  padding: 0px;
  > :first-child {
    height: 18px;
    width: 18px;
    ${spaceSS};
  }
  :hover {
    color: ${props => props.theme.colors.primary};
  }
  :focus,
  :active {
    outline: none;
    box-shadow: none;
  }
  ${fontSizeSS};
  ${fontWeightSS};
  ${colorSS};
`;

interface IProps {
  icon?: React.ReactElement<any>;
  handleClick?: () => void;
  onClick?: () => void;
  disabled?: boolean;
  type?: "button" | "submit" | "reset";
  fontWeight?: number;
  fontSize?: number;
  space?: number;
  color?: string;
}

const TertiaryButtonComponent: React.SFC<IProps> = ({
  type = "button",
  fontWeight = 700,
  fontSize = 0,
  space = 1,
  color = "white",
  disabled = false,
  handleClick,
  onClick,
  icon,
  children,
}) => (
  <TertiaryButton
    type={type}
    disabled={disabled}
    onClick={handleClick || onClick}
    mr={space}
    color={color}
    fontSize={fontSize}
    fontWeight={fontWeight}
  >
    {icon}
    {children}
  </TertiaryButton>
);

export default TertiaryButtonComponent;
