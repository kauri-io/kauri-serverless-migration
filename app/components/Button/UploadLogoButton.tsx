import * as React from "react";
import styled from "../../lib/styled-components";
import {
  fontSize as fontSizeSS,
  color as colorSS,
  space,
  BgColorProps,
} from "styled-system";

interface IButtonProps {
  bg?: string;
  height?: string;
  width?: string;
  fontSize?: number;
  color?: string;
  disabled?: boolean;
}

const UploadLogoButton = styled<IButtonProps & BgColorProps, "button">(
  "button"
)`
  border: 1px solid
    ${({
      theme: {
        colors: { primary },
      },
    }) => primary};
  border-radius: 4px;
  background: ${props =>
    props.bg ? `url(${props.bg}) center center` : "transparent"};
  background-size: cover;
  cursor: ${props => (props.disabled ? "not-allowed" : "pointer")};
  height: ${props => props.height};
  width: ${props => props.width};
  padding: 0;
  text-transform: uppercase;
  opacity: ${props => (props.disabled ? "0.5" : "1")};
  :hover {
    border: 2px solid
      ${({
        theme: {
          colors: { primary },
        },
      }) => primary};
  }
  font-weight: ${props => props.theme.fontWeight[2]};
  ${fontSizeSS};
  ${colorSS};
`;

const Overlay = styled.div`
  background: rgba(0, 0, 0, 0.4);
  display: flex;
  flex-direction: column;
  flex: 1;
  align-items: center;
  justify-content: center;
  height: 100%;

  > :first-child {
    height: 18px;
    width: 18px;
    ${space};
  }
`;

interface IProps extends IButtonProps {
  className?: string;
  handleClick?: () => void;
  onClick?: () => void;
  disabled?: boolean;
  space?: number;
  text?: string;
}

const UploadLogoButtonComponent: React.FunctionComponent<IProps> = ({
  className,
  bg,
  fontSize = 0,
  color = "textPrimary",
  height = "100px",
  width = "100px",
  text = "Logo",
  handleClick,
  onClick,
  children,
  disabled,
}) => (
  <UploadLogoButton
    className={className}
    bg={bg}
    height={height}
    width={width}
    disabled={disabled}
    onClick={onClick || handleClick}
    color={color}
    fontSize={fontSize}
    type="button"
  >
    <Overlay>
      <img src="https://png.icons8.com/color/50/000000/upload.png" />
      {text || children}
    </Overlay>
  </UploadLogoButton>
);

export default UploadLogoButtonComponent;
