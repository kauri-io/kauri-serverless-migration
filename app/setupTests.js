import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
