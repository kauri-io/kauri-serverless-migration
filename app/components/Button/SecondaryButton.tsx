import * as React from "react";
import styled from "styled-components";
import {
  SpaceProps,
  BackgroundProps,
  FontSizeProps,
  FontWeightProps,
  ColorProps,
} from "styled-system";
import { BaseButtonCss } from "./PrimaryButton";

interface ISecondaryButtonProps {
  border: string;
  borderHover: string;
  width: string | undefined;
}

const SecondaryButton = styled<
  ISecondaryButtonProps &
    SpaceProps &
    BackgroundProps &
    FontSizeProps &
    FontWeightProps &
    ColorProps,
  "button"
>("button")`
  ${BaseButtonCss};
  box-shadow: 0px 0px 0px 1px ${props => props.theme.colors[props.border]};
  :hover {
    box-shadow: 0px 0px 0px 2px
      ${props => props.theme.colors[props.borderHover]};
  }
`;

interface IProps {
  icon?: React.ReactElement<any>;
  onClick?: void | (() => void);
  handleClick?: () => void;
  disabled?: boolean;
  fontWeight?: number;
  fontSize?: number;
  width?: string;
  color?: string;
  space?: number;
  text?: string;
  bg?: string;
  border?: string;
  borderHover?: string;
}

const SecondaryButtonComponent: React.SFC<IProps> = ({
  width,
  border = "white",
  borderHover = "primary",
  bg = "transparent",
  fontWeight = 700,
  fontSize = 0,
  color = "white",
  space = 2,
  onClick,
  handleClick,
  text = "",
  children,
  icon,
  disabled,
}) => (
  <SecondaryButton
    type="button"
    width={width}
    disabled={disabled}
    onClick={onClick || handleClick}
    border={border}
    borderHover={borderHover}
    mr={space}
    bg={bg}
    color={color}
    fontSize={fontSize}
    fontWeight={fontWeight}
  >
    {icon}
    {text || children}
  </SecondaryButton>
);

export default SecondaryButtonComponent;
