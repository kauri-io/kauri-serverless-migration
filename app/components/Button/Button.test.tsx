import PrimaryButton from './PrimaryButton'
import SecondaryButton from './SecondaryButton'
import TertiaryButton from './TertiaryButton'
import AddMemberButton from './AddMemberButton'
import MediumImportButton from './MediumImportButton'
import UploadLogoButton from './UploadLogoButton'
import AddTagButton from './AddTagButton'
import { shallowWithTheme } from '../../setupTests'

describe('components/PrimaryButton', () => {
    it('snapshot should match', () => {
        const wrapper = shallowWithTheme(<PrimaryButton>Test</PrimaryButton>)
        expect(wrapper).toMatchSnapshot()
    })
})

describe('components/SecondaryButton', () => {
    it('snapshot should match', () => {
        const wrapper = shallowWithTheme(
            <SecondaryButton>Test</SecondaryButton>
        )
        expect(wrapper).toMatchSnapshot()
    })
})

describe('components/TertiaryButton', () => {
    it('snapshot should match', () => {
        const wrapper = shallowWithTheme(<TertiaryButton>Test</TertiaryButton>)
        expect(wrapper).toMatchSnapshot()
    })
})

describe('components/AddMemberButton', () => {
    it('snapshot should match', () => {
        const wrapper = shallowWithTheme(
            <AddMemberButton>Test</AddMemberButton>
        )
        expect(wrapper).toMatchSnapshot()
    })
})

describe('components/MediumImportButton', () => {
    it('snapshot should match', () => {
        const wrapper = shallowWithTheme(<MediumImportButton />)
        expect(wrapper).toMatchSnapshot()
    })
})

describe('components/UploadLogoButton', () => {
    it('snapshot should match', () => {
        const wrapper = shallowWithTheme(
            <UploadLogoButton>Test</UploadLogoButton>
        )
        expect(wrapper).toMatchSnapshot()
    })
})

describe('components/AddTagButton', () => {
    it('snapshot should match', () => {
        const wrapper = shallowWithTheme(
            <AddTagButton bg="https://test.com/image.png" />
        )
        expect(wrapper).toMatchSnapshot()
    })
})
