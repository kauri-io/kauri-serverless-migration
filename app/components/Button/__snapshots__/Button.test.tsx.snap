// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AddMemberButton snapshot should match 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`components/AddTagButton snapshot should match 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`components/MediumImportButton snapshot should match 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`components/PrimaryButton snapshot should match 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`components/SecondaryButton snapshot should match 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`components/TertiaryButton snapshot should match 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`components/UploadLogoButton snapshot should match 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
